sdk/resourcemanager/azure-resourcemanager-network/src/main/java/com/azure/resourcemanager/network/models/VpnSite.java

```diff
@@ -0,0 +1,170 @@
+
+package com.azure.resourcemanager.network.models;
+
+import com.azure.core.annotation.Fluent;
+import com.azure.resourcemanager.network.NetworkManager;
+import com.azure.resourcemanager.network.fluent.models.VpnSiteInner;
+import com.azure.resourcemanager.network.fluent.models.VpnSiteLinkInner;
+import com.azure.resourcemanager.resources.fluentcore.arm.models.GroupableResource;
+import com.azure.resourcemanager.resources.fluentcore.arm.models.Resource;
+import com.azure.resourcemanager.resources.fluentcore.model.Appliable;
+import com.azure.resourcemanager.resources.fluentcore.model.Creatable;
+import com.azure.resourcemanager.resources.fluentcore.model.Refreshable;
+import com.azure.resourcemanager.resources.fluentcore.model.Updatable;
+
+import java.util.List;
+
+/**
+ * Entry point for Virtual Network management API in Azure.
+ */
+@Fluent
+public interface VpnSite
+    extends GroupableResource<NetworkManager, VpnSiteInner>, Refreshable<VpnSite>, Updatable<VpnSite.Update> {
+
+    /** @return returns true if current VPN site is a security site. */
+    boolean isSecuritySite();
+
+    /** @return the AddressPrefixes value of VPN site. */
+    List<String> addressPrefixes();
+
+    /** @return the virtualWan property of VPN site. */
+    VirtualWan virtualWan();
+
+    /** @return the vpnSiteLinks property of VPN site. */
+    List<VpnSiteLinkInner> vpnSiteLinks();
+
+    /** @return the o365Policy property of VPN site. */
+    O365PolicyProperties o365Policy();
+
+    /** @return the device property of VPN site. */
+    DeviceProperties device();
+
+    /** The entirety of a VPN definition. */
+    interface Definition
+        extends DefinitionStages.Blank,
+        DefinitionStages.WithGroup,
+        DefinitionStages.WithCreate {
+    }
+
+    /** Grouping of VPN definition stages. */
+    interface DefinitionStages {
+        /**
+         * The first stage of a VPN definition.
+         */
+        interface Blank extends GroupableResource.DefinitionWithRegion<WithGroup> {
+        }
+
+        /**
+         * The stage of the VPN definition allowing to specify the resource group.
+         */
+        interface WithGroup extends GroupableResource.DefinitionStages.WithGroup<DefinitionStages.WithCreate> {
+        }
+
+        /**
+         * The stage of a route VPN definition which contains all the minimum required inputs for the resource to be
+         * created (via {@link VpnSite.DefinitionStages.WithCreate#create()}), but also allows for any other optional settings to be specified.
+         */
+        interface WithCreate extends Creatable<VpnSite>, DefinitionWithTags<WithCreate> {
+
+            /**
+             * Set the virtualWan property: The id of VirtualWAN to which the vpn belongs.
+             *
+             * @param subResourceId the id of virtualWan value to set.
+             * @return the next stage of the definition
+             */
+            WithCreate withVirtualWan(String subResourceId);
```

Comment:
Better to separate these `withxx` into different stages.

And some of them seems mandatory, e.g. VirtualWanId, deviceVendor, vpnSiteLinks. Some are not, e.g. addressSpace.